From 65e5ba3c6b5cd0d80e9bb1740c275329faaee145 Mon Sep 17 00:00:00 2001
From: ChromeOS Toolchain Worker
 <crostc-worker@crostc-chrotomation.iam.gserviceaccount.com>
Date: Fri, 14 Jun 2024 21:02:42 +0000
Subject: [PATCH] llvm-8.0-clang-executable-detection.v4

This is to fix crbug/591436, which is a blocker on using clang for
simple-chrome.

patch.cherry: false
patch.metadata.author: Jordan R Abrahams-Whitehead <ajordanr@google.com>
patch.platforms: chromiumos
patch.version_range.from: 614150
patch.version_range.until: null

Change-Id: Id0092c18aecdaf34abdcc6285c555193f71e3171
---
 llvm/lib/Support/Unix/Path.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/lib/Support/Unix/Path.inc b/llvm/lib/Support/Unix/Path.inc
index 3ed56ef01ac9..2615f68fe272 100644
--- a/llvm/lib/Support/Unix/Path.inc
+++ b/llvm/lib/Support/Unix/Path.inc
@@ -257,7 +257,7 @@ std::string getMainExecutable(const char *argv0, void *MainAddr) {
     defined(__managarm__)
   char exe_path[PATH_MAX];
   const char *aPath = "/proc/self/exe";
-  if (sys::fs::exists(aPath)) {
+  if (false && sys::fs::exists(aPath)) {
     // /proc is not always mounted under Linux (chroot for example).
     ssize_t len = ::readlink(aPath, exe_path, sizeof(exe_path));
     if (len < 0)
-- 
2.55.0.979.g7e5102b832-goog

